src/machines/systemIO/systemIOMachineDesktop.ts

```diff
@@ -0,0 +1,233 @@
+import {
+  createNewProjectDirectory,
+  getProjectInfo,
+  mkdirOrNOOP,
+  readAppSettingsFile,
+  renameProjectDirectory,
+} from '@src/lib/desktop'
+import {
+  doesProjectNameNeedInterpolated,
+  getNextFileName,
+  getNextProjectIndex,
+  getUniqueProjectName,
+  interpolateProjectNameWithIndex,
+} from '@src/lib/desktopFS'
+import type { Project } from '@src/lib/project'
+import { systemIOMachine } from '@src/machines/systemIO/systemIOMachine'
+import type { SystemIOContext } from '@src/machines/systemIO/utils'
+import {
+  NO_PROJECT_DIRECTORY,
+  SystemIOMachineActors,
+} from '@src/machines/systemIO/utils'
+import { fromPromise } from 'xstate'
+import type { AppMachineContext } from '@src/lib/types'
+
+export const systemIOMachineDesktop = systemIOMachine.provide({
+  actors: {
+    [SystemIOMachineActors.readFoldersFromProjectDirectory]: fromPromise(
+      async ({ input: context }: { input: SystemIOContext }) => {
+        const projects = []
+        const projectDirectoryPath = context.projectDirectoryPath
+        if (projectDirectoryPath === NO_PROJECT_DIRECTORY) {
+          // TODO
```

Comment:
I don't think there's a TODO here right, this is the expected and tested behavior with no project dir set

Reply:
Correct, I was thinking if I should do something more sophisticated than this but I think this is fair. If we bork the project directory show an empty array of projects.